File: euler008.py
for _ in range(int(input())):
    max_ = 0
    k, n = input().split(' ')
    k, n = int(k), int(n)
    N = input()
    for i in range(k - n + 1):
        temp = 1
        for j in range(n):
            temp *= int(N[i + j])
        if max_ < temp:
            max_ = temp
    print(max_)

File: Euler008.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int t, n, k, max;
    unsigned long long int temp;
    string N;
    cin >> t;
    while (t--)
    {
        max = 0;
        cin >> k >> n >> N;
        for (int i = 0; i <= k - n; i++)
        {
            temp = 1;
            for (int j = 0; j < n; j++)
                temp *= N[i + j] - '0';
            if (temp > max)
                max = temp;
        }
        cout << max << endl;
    }
}